File: westcoast/src/main/java/frc/robot/maps/RobotMap.java
package frc.robot.maps;

import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
import edu.wpi.first.wpilibj.DigitalInput;
import edu.wpi.first.wpilibj.SpeedController;
import edu.wpi.first.wpilibj.controller.PIDController;
import edu.wpi.first.wpilibj.controller.ProfiledPIDController;

public interface RobotMap {
    ProfiledPIDController makeLLPIDs();

    ProfiledPIDController makePTPIDs();

    PIDController makeDrivePIDs();

    PIDController makeAlignPIDs();

    PIDController makeForwardCAlignPIDs();

    PIDController makeTurnCAlignPIDs();

    PIDController makeStayOnTargetPIDs();

    boolean getReversed();

    double getLLMult();

    double getPTMult();

    DigitalInput getIndexSensor();

    DigitalInput getMagSensor();

    DigitalInput getShooterSensor();

    WPI_TalonSRX getMainShooter();

    WPI_TalonSRX getSecondaryShooter();

    SpeedController getIntakeMotor();

    SpeedController getIndexMotor();

    SpeedController getMagMotor();
}

File: westcoast/src/main/kotlin/frc/robot/maps/RobotMap.kt
package frc.robot.maps

import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX
import edu.wpi.first.wpilibj.DigitalInput
import edu.wpi.first.wpilibj.SpeedController
import edu.wpi.first.wpilibj.controller.PIDController
import edu.wpi.first.wpilibj.controller.ProfiledPIDController

interface RobotMap {
  fun makeLLPIDs(): ProfiledPIDController
  fun makePTPIDs(): ProfiledPIDController
  fun makeDrivePIDs(): PIDController
  fun makeAlignPIDs(): PIDController
  fun makeForwardCAlignPIDs(): PIDController
  fun makeTurnCAlignPIDs(): PIDController
  fun makeStayOnTargetPIDs(): PIDController
  val reversed: Boolean
  val lLMult: Double
  val pTMult: Double
  val indexSensor: DigitalInput?
  val magSensor: DigitalInput?
  val shooterSensor: DigitalInput?
  val mainShooter: WPI_TalonSRX?
  val secondaryShooter: WPI_TalonSRX?
  val intakeMotor: SpeedController?
  val indexMotor: SpeedController?
  val magMotor: SpeedController?
}